header.go

```diff
@@ -98,13 +103,82 @@ func (r *Reader) ReadHeader() error {
 		return err
 	}
 
-	r.Header.SyncMarker = string(marker)
+	r.Header.SyncMarker = marker
```

Comment:
we don't just have tiny sequencefiles in tests, my round trip tests write out rather large sequence files. And I just added some more, smaller keys, but more of them. I did address the issue though!